ten-simple-rules-dockerfiles.Rmd

```diff
@@ -398,7 +404,9 @@ LABEL maintainer='D. Nüst <daniel.nuest@uni-muenster.de>' \
 'https://nuest.github.io/ten-simple-rules-dockerfiles/ten-simple-rules-dockerfiles.pdf' \
   org.opencontainers.image.version='1.0.0'
 
-# build-time variable with a default, set it in `docker build` as follows:
+# This ARG instruction adds a build-time variable with the default value 'unspecified'
+# Set it during `docker build` as follows to insert the current date and time into the 
+# label 'org.opencontainers.image.created':
 #   --build-arg BUILD_DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
```

Comment:
This way it is clear that you can set a default value. Maybe use something else than "undefined", but... `development-version` ?

Reply:
If I were reviewing this Dockerfile, I'd say "if you are building the container, and we always have a reliable way to state the date in the dockerfile, it should not be exposed as a build arg, but always set in the container programatically." So I think the `BUILD_DATE` arg is possibly a bad example. A more appropriate example might be `BUILD_TYPE` or something like that, and then the default would be development:

```
ARG BUILD_TYPE development
```
because if you set `BUILD_DATE` to be an arg and rely on the user to use date() on the command line, it's almost undoubtedly going to be forgotten and the arg will be useless because we'll never know the date.

Reply:
I understand. I also thought about this as something that might not be used often, and might be easily forgotten unless automated builds/Makefile etc. are used.

Do you suggest to keep it in?

What would be an example for the non-default value? `BUILD_TYPE=preprint-submission-v1` ? Or would that better be stored in a label?

Reply:
production would be the arg used when the container is deployed /versioned. It would likely drive differences in the container for development vs production.

Reply:
I've removed the `.created` label and also the `ARG` example, see further comments in #70